src/mlpack/core/data/tokenizer/strtok.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file strtok.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of Strtok class which tokenizes using single character.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_STRTOK_HPP
+#define MLPACK_CORE_DATA_STRTOK_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "mlpack/core/boost_backport/boost_backport_string_view.hpp"
+
+namespace mlpack {
+namespace data {
+/**
+ * A simple Strtok class
+ */
+class Strtok
```

Comment:
I suggest to replace the class name. Try to find some name that indicates the tokenization type.

Reply:
Would CharSplit work ?

Reply:
Yes, I think `CharSplit` suits well.